fvtest/porttest/omrsockTest.cpp

```diff
@@ -0,0 +1,138 @@
+ */ 
+int32_t
+start_server(struct OMRPortLibrary *portLibrary, const char *testName, const char *addrStr, const char *port, int32_t family, omrsock_socket_t *serverSocket, omrsock_sockaddr_t serverAddr) 
+{
+    return OMRPORT_ERROR_NOTEXIST;
+}
+
+/**
+ * Create the client socket, and then connect to the server.
+ *  
+ * @param[in] portLibrary
+ * @param[in] testname The name of the calling test.
+ * @param[in] addrStr The address of the server.
+ * @param[in] port The server port.
+ * @param[in] family Socket address family wanted.
+ * @param[in] sessionClientSocket A pointer to the client socket. 
+ * @param[in] sessionClientAddr The socket address of the client created. 
+ * 
+ * @return 0 on success, return an error otherwise.
+ */ 
+int32_t
+connectClientToServer(struct OMRPortLibrary *portLibrary, const char* testName, const char *addrStr, const char *port, int32_t family, omrsock_socket_t *sessionClientSocket, omrsock_sockaddr_t sessionClientAddr) 
+{
+    return OMRPORT_ERROR_NOTEXIST;
+}
+
+/**
+ * Test if port library is properly set up to run OMRSock tests. 
+ *
+ * This will iterate through all OMRSock functions in @ref omrsock.c and simply check 
+ * that the location of the function is not NULL.
+ * 
+ * @return TEST_PASS on success, TEST_FAIL on error.
+ *
+ * @note Errors such as an OMRSock function call with a NULL address will be reported.
+ */
+TEST(PortSockTest, sock_test0)
+{
+    OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+    const char *testName = "omrsock_test0";
+}
+
+/**
+ * Test the omrsock per thread buffer.
+ * 
+ * In this test, per thread buffer related functions set up a buffer to store
+ * information such as the hints structures created in @ref omrsock_getaddrinfo_create_hints.
+ *
+ * @return TEST_PASS on success, TEST_FAIL on error.
+ */
+TEST(PortSockTest, sock_test1)
+{
+    OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+    const char *testName = "omrsock_test1";
+}
+
+/**
+ * Test @ref omrsock_getaddrinfo_create_hints, @ref omrsock_getaddrinfo, 
+ * @ref omrsock_freeaddrinfo and all functions that extract details from
+ * @ref omrsock_getaddrinfo results.
+ * 
+ * In this test, the relevant variables are passed into  
+ * @ref omrsock_getaddrinfo_create_hints. The generated hints are passed into 
+ * @ref omrsock_getaddrinfo. The results generated are used to create a socket.
+ *
+ * Address families tested will include IPv4, IPv6(if supported on machine), 
+ * unspecified family. Socket types tested will include Stream and Datagram.
+ *
+ * @return TEST_PASS on success, TEST_FAIL on error.
+ *
+ * @note Errors such as failed function calls, and/or returning the wrong family or 
+ * socket type from @ref omrsock_getaddrinfo compared to the ones passed into hints, 
+ * will be reported.
+ */
+TEST(PortSockTest, sock_test2)
+{
+    OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+    const char *testName = "omrsock_test2";
```

Comment:
You do not need to define the test name, gtest gives us a facility already.
https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#getting-the-current-tests-name

Reply:
Sounds good, thank you Robert.